src/utils/router_state.rs

```diff
@@ -0,0 +1,420 @@
+#[derive(Debug, Copy, Clone)]
+pub enum Aircraft {
+    ///Cargo aircraft
+    Cargo,
+}
+/// List of vertiport nodes for routing
+pub static NODES: OnceCell<Vec<Node>> = OnceCell::new();
+/// Cargo router
+pub static ARROW_CARGO_ROUTER: OnceCell<Router> = OnceCell::new();
+
+static ARROW_CARGO_CONSTRAINT: f32 = 75.0;
+/// SF central location
+pub static SAN_FRANCISCO: Location = Location {
+    latitude: OrderedFloat(37.7749),
+    longitude: OrderedFloat(-122.4194),
+    altitude_meters: OrderedFloat(0.0),
+};
+
+/// Time to block vertiport for cargo loading and takeoff
+pub const LOADING_AND_TAKEOFF_TIME_MIN: f32 = 10.0;
+/// Time to block vertiport for cargo unloading and landing
+pub const LANDING_AND_UNLOADING_TIME_MIN: f32 = 10.0;
+/// Average speed of cargo aircraft
+pub const AVG_SPEED_KMH: f32 = 60.0;
+
+/// Creates all possible flight plans based on the given request
+/// * `vertiport_depart` - Departure vertiport - svc-storage format
+/// * `vertiport_arrive` - Arrival vertiport - svc-storage format
+/// * `departure_time` - Departure time
+/// * `arrival_time` - Arrival time
+/// * `aircrafts` - Aircrafts serving the route and vertiports
+/// # Returns
+/// A vector of flight plans
+pub fn get_possible_flights(
+    vertiport_depart: Vertiport,
+    vertiport_arrive: Vertiport,
+    departure_time: Option<Timestamp>,
+    arrival_time: Option<Timestamp>,
+    aircrafts: Vec<Vehicle>,
+) -> Result<Vec<FlightPlanData>, String> {
+    info!("Finding possible flights");
+    //1. Find route and cost between requested vertiports
+    info!("[1/5]: Finding route between vertiports");
+    if !is_router_initialized() {
+        error!("Router not initialized");
+        return Err("Router not initialized".to_string());
+    }
+    let (route, cost) = get_route(RouteQuery {
+        from: get_node_by_id(&vertiport_depart.id)?,
+        to: get_node_by_id(&vertiport_arrive.id)?,
+        aircraft: Aircraft::Cargo,
+    })?;
+    debug!("Route: {:?}", route);
+    debug!("Cost: {:?}", cost);
+    if route.is_empty() {
+        error!("No route found");
+        return Err("Route between vertiports not found".to_string());
+    }
+
+    //2. calculate blocking times for each vertiport and aircraft
+    info!("[2/5]: Calculating blocking times");
+    let block_departure_vertiport_minutes = LOADING_AND_TAKEOFF_TIME_MIN;
+    let block_arrival_vertiport_minutes = LANDING_AND_UNLOADING_TIME_MIN;
+    let block_aircraft_minutes = estimate_flight_time_minutes(cost, Aircraft::Cargo);
+    debug!(
+        "Estimated flight time in minutes: {}",
+        block_aircraft_minutes
+    );
+
+    //3. check vertiport schedules and flight plans
+    info!("[3/5]: Checking vertiport schedules and flight plans");
+    const SAMPLE_CAL: &str =
+        "DTSTART:20221020T180000Z;DURATION:PT1H\nRRULE:FREQ=WEEKLY;BYDAY=MO,TU,WE,TH,FR";
+    let departure_vertiport_schedule = Calendar::from_str(SAMPLE_CAL).unwrap(); //TODO get from DB
```

Comment:
eventually this will take a non-sample CAL string and isn't necessarily guaranteed to be well formed

handle unwraps

Reply:
yes, everything with //todo is just a temp code and will be handled when we have real calendars coming from db